polkadot/xcm/src/v3/mod.rs

```diff
@@ -60,13 +63,28 @@ pub const VERSION: super::Version = 3;
 /// An identifier for a query.
 pub type QueryId = u64;
 
-#[derive(Derivative, Default, Encode, Decode, TypeInfo)]
+#[derive(Derivative, Default, Encode, TypeInfo)]
 #[derivative(Clone(bound = ""), Eq(bound = ""), PartialEq(bound = ""), Debug(bound = ""))]
 #[codec(encode_bound())]
-#[codec(decode_bound())]
 #[scale_info(bounds(), skip_type_params(Call))]
 pub struct Xcm<Call>(pub Vec<Instruction<Call>>);
 
+const MAX_INSTRUCTIONS_TO_DECODE: u32 = 100;
```

Comment:
Could we not make this an extra generic parameter? We could set some default value and then it would also not be breaking.

Reply:
Do we want to allow for `Xcm` instances with other possible values for this? I didn't think about it. I don't know if we do.

The only thing about using `BoundedVec` directly are the changes to the API. All the methods that become fallible and so on.
Changing this to a generic is independent of using `BoundedVec`

Reply:
I think Basti meant using const generics, i.e. `Xcm<Call, const MAX_LEN: u32 = 100>`

Reply:
I could do that, but do we want to allow stuff like `Xcm<RuntimeCall>` and `Xcm<RuntimeCall, 200>` to coexist?

Reply:
If they do coexist, then they're incompatible types. Nobody should be modifying the const generic parameter without a very good reason.

Reply:
Then I don't see the benefit of the const generic vs the constant

Reply:
> Do we want to allow for Xcm instances with other possible values for this?  

Yes, if it doesn't cost anything in terms of API flexibility.

We'll need to be careful to ensure anything which works with the `Xcm` type is generic over this new param.